src/views/virtualmachines/details/components/details/hooks/useVMIPod.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  modelToGroupVersionKind,
+  PodModel,
+  VirtualMachineInstanceModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseVMIPod = {
+  vmi: V1VirtualMachineInstance;
+  pods: K8sResourceCommon[];
+  loaded: boolean;
+  error: any;
+};
+
+const useVMIPod = (vmName: string, vmNamespace: string): UseVMIPod => {
```

Comment:
This name makes me think that you're using the pod for a VMI. I would consider `useVMIAndPodsForVM`.